Solved Issue #68 [FEATURE] Tablet Mode for Teams Section [GSSoC'23]

vishnu9865 opened the pull request:
## Description
Feat: enable Tablet Mode for Teams Section

## Related Issue(s)

Closes #68 

## Proposed Changes
 - Edited _app.js (temp fix for minimizing conflicts with other issues).
 - Created Developer.module.css.
 - Modified CSS for Tablet Mode.
 - Fixed layout bugs in Tablet Mode.
 - Edited CSS to ease transition to TailwindCSS.

## Screenshots (if applicable)
![image](https://github.com/Deveimer/vovoca-frontend/assets/91585213/3dde53ed-abef-4b96-a214-bf004f7adbda)

## Checklist
- [X] I have tested the changes locally and they are working as expected.
- [X] I have added any necessary documentation or updated existing documentation.
- [X] I have reviewed the code changes to ensure they adhere to the project's coding standards.
- [X] I have added appropriate labels to this pull request.

vishnu9865 committed:
feat: enable Tablet Mode for Teams Section
 - Created Developer.module.css.
 - Modified CSS for Tablet Mode.
 - Fixed layout bugs in Tablet Mode.
 - Edited CSS to ease transition to TailwindCSS.

kaustubhai commented on pages/_app.js:
```diff
@@ -12,7 +12,7 @@ import ScrollButton from '../components/scrollToTop/ScrollButton';
 const Tablet = () => {
   const isTablet = useMediaQuery({ minWidth: 601, maxWidth: 1226 });
-  return isTablet;
+  return isTablet && false;
```
This will return false for every page. Use an argument in `Tablet` function to determine page and then return false for `/developers` only

vishnu9865 replied:
@kaustubhai Ok sir,
Working on it now 🚀 🚀

kaustubhai commented:
![image](https://github.com/Deveimer/vovoca-frontend/assets/47121025/7bacce12-ad8a-4404-a9eb-f7087d9ac2c1)

This is not an accurate design @vishnu9865. You will have to push a little harder, this is the first time we are developing a page for tablet so you will have to adjust minor details of components being used in every page as well

vishnu9865 commented:
> ![image](https://user-images.githubusercontent.com/47121025/243119748-7bacce12-ad8a-4404-a9eb-f7087d9ac2c1.png)
> 
> This is not an accurate design @vishnu9865. You will have to push a little harder, this is the first time we are developing a page for tablet so you will have to adjust minor details of components being used in every page as well

@kaustubhai Sir,
Sorry sir, I thought I should not tamper with navbar code as my PR may have conflicts with any other contributor working on Navbar alignment or any navbar related issue.
Is it okay for me to work on navbar ? If so I am very happy solve the issue in this PR.
Thank you for your time with this PR.
Regards,
Jeyavishnu S @vishnu9865

vishnu9865 committed:
fix: modify tablet function
 - Tablet function now takes nextjs router object as argument.
 - Anyone wanting to enable a page can add to the conditions in the return statement easily.

vishnu9865 committed:
Merge branch 'Deveimer:Development' into VOVOCA-68

vishnu9865 committed:
fix: Remove console logs
 - Removed uneccessary console logs in the code.

vishnu9865 committed:
Merge branch 'VOVOCA-68' of https://github.com/vishnu9865/vovoca-frontend into VOVOCA-68

kaustubhai commented:
> @kaustubhai Sir, Sorry sir, I thought I should not tamper with navbar code as my PR may have conflicts with any other contributor working on Navbar alignment or any navbar related issue. Is it okay for me to work on navbar ? If so I am very happy solve the issue in this PR. Thank you for your time with this PR. Regards, Jeyavishnu S @vishnu9865

Yeah yeah please go ahead. I think it was mentioned in the issue itself that this is the first page that was being migrated to tablet mode. Kudos for thinking for conflicts but, you had a go-ahead already to work on that

vishnu9865 commented:
Ok sir, working on it 🚀🚀🚀

vishnu9865 committed:
fix: edit navbar for tablet mode.
 - navbar changes layout to a flex container with space-between.
 - Mobile navbar becomes visible at max-width: 775px
 - Changes to VOVOCA icon in navbar for better layout and responsiveness.

vishnu9865 committed:
fix: further modify Navbar
 - additional bug fixed for the tablet mode navbar.

vishnu9865 commented:
@kaustubhai Sir,
I have made some changes to the Navbar code.
Please feel free to ask for any changes required and sorry for any inconveniences caused. 
Regards,
Jeyavishnu @vishnu9865 
![image](https://github.com/Deveimer/vovoca-frontend/assets/91585213/6d8278cb-a86f-4912-bc19-3935657aeb7f)

kaustubhai commented:
Sharing few pointers with you, I think it will look better then:
1. Remove the Home link from navbar for tab and desktop
2. Decrease the Dashboard button size (Can remove the icon and decrease the font-size) only for tab
3. Make button-group right-aligned and vovoca to the left as it is in desktop

vishnu9865 commented:
@kaustubhai Ok sir, working on it.

vishnu9865 committed:
fix: update navbar design
 - Remove Home button
 - Remove Flex layout and make navbar group right aligned.

vishnu9865 committed:
fix: remove dashboard icon in tablet mode
 - edited CSS rules to remove icon in tablet mode.

vishnu9865 commented:
@kaustubhai Sir, 
I made the following changes. 👍
![image](https://github.com/Deveimer/vovoca-frontend/assets/91585213/13cbfefe-843a-4ff8-a919-8d4f13db85d4)

kaustubhai commented:
In review @vishnu9865

vishnu9865 commented:
> In review @vishnu9865

@kaustubhai Sir, ??
